"java.io.tmpdir" is set to "/tmp" by default on MOE

"java.io.tmpdir" is set to "/tmp" by default on MOE. 
https://github.com/multi-os-engine/libcore/blob/eae58129e020f95e9b4a506cea4cc90cc21b09a3/luni/src/main/java/java/lang/System.java#L851-L853
However, on iOS this path is invalid as a temporary directory.


This leads to errors like this:
createTempDirectory hangs indefinitly and createTempFile throws a error like:
```
java.io.FileNotFoundException: /tmp/random-name9074725035171402104.irrelevant: open failed: EPERM (Operation not permitted)
[...]
Caused by: java.io.FileNotFoundException: /tmp/random-name9074725035171402104.irrelevant: open failed: EPERM (Operation not permitted)
	at libcore.io.IoBridge.open(IoBridge.java:462)
	at java.io.RandomAccessFile.<init>(RandomAccessFile.java:115)
	at desugar.sun.nio.fs.DesugarSeekableByteChannel.create(DesugarSeekableByteChannel.java:44)
	at desugar.sun.nio.fs.DesugarLinuxFileSystemProvider.newByteChannel(DesugarLinuxFileSystemProvider.java:176)
	at desugar.sun.nio.fs.DesugarLinuxFileSystemProvider.newByteChannel(DesugarLinuxFileSystemProvider.java:59)
	at java.nio.file.Files.newByteChannel(Files.java:372)
	at java.nio.file.Files.createFile(Files.java:649)
	at java.nio.file.TempFileHelper.create(TempFileHelper.java:137)
	at java.nio.file.TempFileHelper.createTempFile(TempFileHelper.java:160)
	at java.nio.file.Files.createTempFile(Files.java:914)
	at com.fittonia.chigginblazzder.IOSLauncher$$ExternalSyntheticApiModelOutline0.m(R8$$SyntheticClass)
	at com.fittonia.chigginblazzder.IOSLauncher.extractFileFromResources(IOSLauncher.java:32)
	at com.fittonia.chigginblazzder.IOSLauncher.createApplication(IOSLauncher.java:56)
	... 5 more
Caused by: android.system.ErrnoException: open failed: EPERM (Operation not permitted)
	at libcore.io.Posix.open(Native Method)
	at libcore.io.BlockGuardOs.open(BlockGuardOs.java:187)
	at libcore.io.IoBridge.open(IoBridge.java:448)
	... 17 more
```

--- a/moe.apple/moe.core.native/moe.sdk/src/MOE.mm
+++ b/moe.apple/moe.core.native/moe.sdk/src/MOE.mm
@@ -171,6 +171,12 @@ extern "C" int run_moevm(int isDebug, const int jargc, char* const* jargv) {
         [args addObject:[NSString stringWithFormat:@"-Duser.locale=%@", localeStr]];
     }
 
+    NSString* tmpStr =
+        [NSString stringWithFormat:@"-Djava.io.tmpdir=%s", getenv("MOE_TMP_DIR")];
+
+    [args addObject:tmpStr];
+
+
     // Create arguments for classpath
     [args addObject:@CLASSPATH_PREFIX];
     [args addObject:@""];

